ocrd_network/ocrd_network/processing_worker.py

```diff
@@ -0,0 +1,327 @@
+        self.rmq_consumer = None
+
+        # Gets assigned when the `connect_publisher` is called on the worker object
+        # The publisher is connected when the `result_queue` field of the OcrdProcessingMessage is set for first time
+        # Used to publish OcrdResultMessage type message to the queue with name {processor_name}-result
+        self.rmq_publisher = None
+
+    def connect_consumer(self) -> None:
+        self.log.info(f'Connecting RMQConsumer to RabbitMQ server: '
+                      f'{self.rmq_host}:{self.rmq_port}{self.rmq_vhost}')
+        self.rmq_consumer = RMQConsumer(
+            host=self.rmq_host,
+            port=self.rmq_port,
+            vhost=self.rmq_vhost
+        )
+        self.log.debug(f'RMQConsumer authenticates with username: '
+                       f'{self.rmq_username}, password: {self.rmq_password}')
+        self.rmq_consumer.authenticate_and_connect(
+            username=self.rmq_username,
+            password=self.rmq_password
+        )
+        self.log.info(f'Successfully connected RMQConsumer.')
+
+    def connect_publisher(self, enable_acks: bool = True) -> None:
+        self.log.info(f'Connecting RMQPublisher to RabbitMQ server: '
+                      f'{self.rmq_host}:{self.rmq_port}{self.rmq_vhost}')
+        self.rmq_publisher = RMQPublisher(
+            host=self.rmq_host,
+            port=self.rmq_port,
+            vhost=self.rmq_vhost
+        )
+        self.log.debug(f'RMQPublisher authenticates with username: '
+                       f'{self.rmq_username}, password: {self.rmq_password}')
+        self.rmq_publisher.authenticate_and_connect(
+            username=self.rmq_username,
+            password=self.rmq_password
+        )
+        if enable_acks:
+            self.rmq_publisher.enable_delivery_confirmations()
+            self.log.info('Delivery confirmations are enabled')
+        self.log.info('Successfully connected RMQPublisher.')
+
+    # Define what happens every time a message is consumed
+    # from the queue with name self.processor_name
+    def on_consumed_message(
+            self,
+            channel: pika.adapters.blocking_connection.BlockingChannel,
+            delivery: pika.spec.Basic.Deliver,
+            properties: pika.spec.BasicProperties,
+            body: bytes) -> None:
+        consumer_tag = delivery.consumer_tag
+        delivery_tag: int = delivery.delivery_tag
+        is_redelivered: bool = delivery.redelivered
+        message_headers: dict = properties.headers
+
+        self.log.debug(f'Consumer tag: {consumer_tag}, '
+                       f'message delivery tag: {delivery_tag}, '
+                       f'redelivered: {is_redelivered}')
+        self.log.debug(f'Message headers: {message_headers}')
+
+        try:
+            self.log.debug(f'Trying to decode processing message with tag: {delivery_tag}')
+            processing_message: OcrdProcessingMessage = OcrdProcessingMessage.decode_yml(body)
```

Comment:
I would recommend to validate the message before trying to decode it. We have the message schema, which can and should be used for validation.

Reply:
Ah, I see. Then, validate it before processing it.

Reply:
@tdoan2010, it is already validated before processing it. Check inside the constructor of the `OcrdProcessingMessage`. The `decode_yml` method calls the constructor with the decoded data.

Reply:
Checked and resolved.